api/v1/storagecluster_types.go

```diff
@@ -26,13 +26,21 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+// ApiServerKind defines the type of servers
+type ApiServerKind string
+
+const (
+	// ProviderApiServer defines the ocs-to-ocs provider API server
+	ProviderApiServer ApiServerKind = "provider"
```

Comment:
Do we have more than one API server type/kind to support? 
I am not aware of any other API server implementation.

Reply:
As of now, we have only one, But we don't know about the future it will help us to have more deployments in the future. Maybe we can rename the `ApiServers` in the spec to `Deployments`

Reply:
The point is that fact that the API server needs a new deployment is just an implementation detail and we should not expose it in the stroagecluster API (see one of my other comments below for a full explanation of my position)

Reply:
done